.github/workflows/redfield-sanity-check.yml

```diff
@@ -0,0 +1,39 @@
+# Syntax: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+# .NET CLI: https://docs.microsoft.com/dotnet/core/tools/
+# Description: The purpose of this build is to build and test with redfield flag.
+
+name: Redfield Sanity Check
+
+on:
+  push:
+    branches: [ master, develop ]
+  pull_request:
+    branches: [ master, develop ]
+
+jobs:
+  build-test:
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        framework: [netcoreapp3.1,net5.0]
+        include: 
+            - os: ubuntu-latest
+              args: "--filter TestCategory!=WindowsOnly"
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: restore (Base sln)
+      run: dotnet restore ./BASE/Microsoft.ApplicationInsights.sln
+
+    - name: Build
+      run: dotnet build -p:Redfield=True ./BASE/Microsoft.ApplicationInsights.sln --configuration Release --no-restore
+
+    - name: Test
+      id: test1
+      continue-on-error: true
+      run: dotnet test ./BASE/Microsoft.ApplicationInsights.sln --framework ${{ matrix.framework }} --configuration Release --no-build --logger:"console;verbosity=detailed" ${{ matrix.args }}
```

Comment:
Because all the other builds run the full tests, I think for this build we can run only the Redfield tests.
We can filter by test name or introduce a new attribute.

Reply:
There are changes in diagnostic listeners and few tests to include Redfield event source. It would be good to run it against the complete project. For example, Please check changes to tests `EventCounterListener.cs` in this PR.